test/functional/query-builder/locking/query-builder-locking.ts

```diff
@@ -975,7 +975,10 @@ describe("query builder > locking", () => {
                                 .createQueryBuilder(Post, "post")
                                 .leftJoin("post.author", "user")
                                 .setLock("pessimistic_write")
-                                .getOne(),
+                                .getOne()
+                                .should.be.rejectedWith(
```

Comment:
This seems to be breaking the transaction and making `RELEASE SAVEPOINT` unavailable: https://www.cockroachlabs.com/docs/stable/release-savepoint#handling-errors

Because Chai does not forward the error, `Promise.all` will not throw either, making `EntityManager.transaction` think the operation was successful, so it won't roll back.

We should probably make that function fail.

Reply:
Yepp, it's the same thing in the privious test ;) There's this comment: 

https://github.com/typeorm/typeorm/pull/11190/files#diff-0c1916cf8da19e54f4ed20c05df0044acf71abe15bc7db30e4275ba35ec13b17L953